src/btree/bt_handle.c

```diff
@@ -15,6 +15,39 @@ static int __btree_preload(WT_SESSION_IMPL *);
 static int __btree_tree_open_empty(WT_SESSION_IMPL *, bool);
 
 /*
+ * __btree_initialize --
+ *	Initialize the WT_BTREE structure.
+ */
+static void
+__btree_initialize(WT_BTREE *btree, bool closing)
+{
+	uint32_t mask;
+
+	/*
+	 * This function exists as a place to talk about how the WT_BTREE
+	 * structure is initialized (or re-initialized, when the object is
+	 * re-opened). The upper-level handle code currently sets/clears
+	 * information in the WT_BTREE structure, specifically flags, cache
+	 * and eviction information. All of that information must persist
+	 * after the WT_BTREE object is closed, and some of it has already
+	 * been initialized when the object is opened/re-opened.
+	 */
+	if (closing) {
+		/*
+		 * Closing: clear everything except cache/eviction information
+		 * and one LSM flag.
+		 */
+		memset(btree, 0, WT_BTREE_CLEAR_SIZE(btree));
+		F_CLR(btree, ~(WT_BTREE_LSM_PRIMARY | WT_BTREE_NO_EVICTION));
```

Comment:
I think the only way the `WT_BTREE_LSM_PRIMARY` flag can be set here is during `WT_CONNECTION::close` -- I don't think there's any reason to preserve that flag after a btree handle is closed.

Its purpose is for accounting, it could be called something like `WT_BTREE_IGNORE_DIRTY_SIZE` to clarify the intent.